Context adjusted for Kernel 4.9 by Junde Yhi <lmy441900@aosc.xyz>

Original Git commit message:

From bd42a8f1017b0b456ce8dc2fe0e8c2eb7fd3d2c7 Mon Sep 17 00:00:00 2001
From: Donavan Lance <tusklahoma@gmail.com>
Date: Mon, 2 Nov 2015 16:06:40 -0500
Subject: [PATCH] Add multitouch support for Microsoft Type Cover 3

Add preliminary support for Type Cover 4 and Surface Book keyboard

Patch based on code from Felipe Otamendi <https://www.github.com/felipeota>
[buribullet@gmail.com: Add Japanese Type Cover]
Contributor: Kobayasi Hiroaki <buribullet@gmail.com>
Maintainer: Donavan Lance <tusklahoma@gmail.com>

Signed-off-by: Donavan Lance <tusklahoma@gmail.com>
---
--- a/drivers/hid/hid-core.c	2017-07-22 18:05:54.053710159 +0800
+++ b/drivers/hid/hid-core.c	2017-07-22 20:19:55.827225901 +0800
@@ -724,11 +724,7 @@
 		hid->group = HID_GROUP_SENSOR_HUB;
 
 	if (hid->vendor == USB_VENDOR_ID_MICROSOFT &&
-	    (hid->product == USB_DEVICE_ID_MS_TYPE_COVER_PRO_3 ||
-	     hid->product == USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_2 ||
-	     hid->product == USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_JP ||
-	     hid->product == USB_DEVICE_ID_MS_TYPE_COVER_PRO_4_JP ||
-	     hid->product == USB_DEVICE_ID_MS_POWER_COVER) &&
+	    hid->product == USB_DEVICE_ID_MS_POWER_COVER &&
 	    hid->group == HID_GROUP_MULTITOUCH)
 		hid->group = HID_GROUP_GENERIC;
 
--- a/drivers/hid/hid-ids.h	2017-07-22 18:05:54.053710159 +0800
+++ b/drivers/hid/hid-ids.h	2017-07-22 20:21:18.657227789 +0800
@@ -731,6 +731,10 @@
 #define USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_2  0x07e2
 #define USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_JP 0x07dd
 #define USB_DEVICE_ID_MS_TYPE_COVER_PRO_4_JP 0x07e9
+#define USB_DEVICE_ID_MS_TYPE_COVER_3_2  0x07e2
+#define USB_DEVICE_ID_MS_TYPE_COVER_4    0x07e4
+#define USB_DEVICE_ID_MS_TYPE_COVER_4_2  0x07e8
+#define USB_DEVICE_ID_MS_SURFACE_BOOK    0x07cd
 #define USB_DEVICE_ID_MS_POWER_COVER     0x07da
 
 #define USB_VENDOR_ID_MOJO		0x8282
--- a/drivers/hid/usbhid/hid-quirks.c	2017-07-22 18:05:54.053710159 +0800
+++ b/drivers/hid/usbhid/hid-quirks.c	2017-07-22 20:23:33.113897507 +0800
@@ -108,6 +108,10 @@
 	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_2, HID_QUIRK_NO_INIT_REPORTS },
 	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_TYPE_COVER_PRO_3_JP, HID_QUIRK_NO_INIT_REPORTS },
 	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_TYPE_COVER_PRO_4_JP, HID_QUIRK_NO_INIT_REPORTS },
+	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_TYPE_COVER_3_2, HID_QUIRK_NO_INIT_REPORTS },
+	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_TYPE_COVER_4, HID_QUIRK_NO_INIT_REPORTS },
+	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_TYPE_COVER_4_2, HID_QUIRK_NO_INIT_REPORTS },
+	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_SURFACE_BOOK, HID_QUIRK_NO_INIT_REPORTS },
 	{ USB_VENDOR_ID_MICROSOFT, USB_DEVICE_ID_MS_POWER_COVER, HID_QUIRK_NO_INIT_REPORTS },
 	{ USB_VENDOR_ID_MSI, USB_DEVICE_ID_MSI_GT683R_LED_PANEL, HID_QUIRK_NO_INIT_REPORTS },
 	{ USB_VENDOR_ID_NEXIO, USB_DEVICE_ID_NEXIO_MULTITOUCH_PTI0750, HID_QUIRK_NO_INIT_REPORTS },
